src/librustc_mir/interpret/machine.rs

```diff
@@ -318,4 +318,11 @@ pub trait Machine<'mir, 'tcx>: Sized {
         _mem: &Memory<'mir, 'tcx, Self>,
         _ptr: Pointer<Self::PointerTag>,
     ) -> InterpResult<'tcx, u64>;
+
+    fn thread_local_alloc_id(
+        _ecx: &mut InterpCx<'mir, 'tcx, Self>,
+        did: DefId,
+    ) -> InterpResult<'tcx, AllocId> {
+        throw_unsup!(ThreadLocalStatic(did))
```

Comment:
Hm okay the default implementation here makes sense. Maybe errors that are shared between const-prop and const-eval should indeed be in `librustc_middle`'s error type.
However, we should have comments there separating them from the errors *any* instance of the engine can raise (including Miri-the-tool).

Reply:
Seems to me like this should enable unleashed miri to detect constants/statics creating references to TLS statics; could you add a test?

Reply:
Then I'd have to lift the restriction that thread locals are hard-caught by the static checks in https://github.com/rust-lang/rust/blob/4e4d49d60fd696c4036d438292673a2d7fd34519/src/librustc_mir/transform/check_consts/ops.rs#L372

Reply:
If swapping that constant is all it takes, that seems great. :D The plan was anyway to get rid of that constant (https://github.com/rust-lang/const-eval/issues/17#issuecomment-613390356).

Reply:
do you want the bool swap to happen in this PR or should we just wait for the removal PR?

Reply:
Whatever you prefer.

Reply:
let's just wait for the removal PR